src/transactions/TransactionUtils.cpp

```diff
@@ -232,7 +232,7 @@ addBalance(LedgerTxnHeader const& header, LedgerTxnEntry& entry, int64_t delta)
             return true;
         }
 
-        if (header.current().ledgerVersion < 10 && !isAuthorized(entry))
+        if (!isAuthorizedToMaintainLiabilities(entry))
```

Comment:
This doesn't make any sense to me. I see the discussion https://github.com/stellar/stellar-core/pull/2387#discussion_r390052332 (GitHub is giving me a hard time about replying to it....). It should be `!isAuthorized`, because you can't add balance if you are only authorized to maintain liabilities. Same remark applies for this change in `addBuyingLiabilities` and `addSellingLiabilities`: you can't add liabilities if you are only authorized to maintain liabilities. Am I missing something?

Reply:
The balance can change if an existing offer was filled. Similar thing with the add liabilities methods - liabilities can be removed when an offer crosses. The offer can also be deleted.